experimenter/experimenter.py

```diff
@@ -83,75 +84,71 @@ def __init__(self, datasets_dir: str, volumes_dir: str, input_problem_directory=
+            step_counter: The the number of the next step to be worked on after the call to
+                this method is complete.
+            attributes_step_i: The number of the step whose ouptut holds the final version
+                of the attributes
+            target_step_i: The number of the step whose output holds the target data.
         """
         # Creating pipeline
         pipeline_description.add_input(name='inputs')
 
-        # TODO: decide when to use this
-        # # Step 0: Denormalize - not required
-        # denormalizer: PrimitiveBase = index.get_primitive("d3m.primitives.data_transformation.denormalize.Common")
-        # step_0 = PrimitiveStep(primitive_description=denormalizer.metadata.query())
-        # step_0.add_argument(name='inputs', argument_type=ArgumentType.CONTAINER, data_reference='steps.{}.produce'.format(step_counter - 1))
-        # step_0.add_output('produce')
-        # pipeline_description.add_step(step_0)
-        # step_counter += 1
-
-        # Step 1: dataset_to_dataframe
-        step_1 = PrimitiveStep(
-            primitive=d3m_index.get_primitive('d3m.primitives.data_transformation.dataset_to_dataframe.Common'))
-        step_1.add_argument(name='inputs', argument_type=ArgumentType.CONTAINER, data_reference='inputs.0')
-        step_1.add_output('produce')
-        pipeline_description.add_step(step_1)
-        step_counter += 1
-
-        # Step 2: column_parser
-        step_2 = PrimitiveStep(
-            primitive=d3m_index.get_primitive('d3m.primitives.data_transformation.column_parser.DataFrameCommon'))
-        step_2.add_argument(name='inputs', argument_type=ArgumentType.CONTAINER,
-                            data_reference='steps.{}.produce'.format(step_counter - 1))
-        step_2.add_output('produce')
-        pipeline_description.add_step(step_2)
-        step_counter += 1
-
-        # Step 3: Imputer
-        sk_imputer: PrimitiveBase = d3m_index.get_primitive("d3m.primitives.data_preprocessing.random_sampling_imputer.BYU")
-        step_3 = PrimitiveStep(primitive_description=sk_imputer.metadata.query())
-        step_3.add_argument(name='inputs', argument_type=ArgumentType.CONTAINER,
-                            data_reference = 'steps.{}.produce'.format(step_counter - 1))
-        step_3.add_output('produce')
-        pipeline_description.add_step(step_3)
-        step_counter += 1
-
-        # Step 4: extract_columns_by_semantic_types(targets)
-        step_4 = PrimitiveStep(primitive=d3m_index.get_primitive(
-            'd3m.primitives.data_transformation.extract_columns_by_semantic_types.DataFrameCommon'))
-        step_4.add_argument(name='inputs', argument_type=ArgumentType.CONTAINER, data_reference='steps.0.produce')
-        step_4.add_output('produce')
-        step_4.add_hyperparameter(name='semantic_types', argument_type=ArgumentType.VALUE,
+        # dataset_to_dataframe step
+        step_counter = add_pipeline_step(
```

Comment:
This `add_pipeline_step` helper is useful for adding basic steps that only need an `inputs` argument.